src/System.Text.Formatting/System/Text/Parsing/InvariantParser_long.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Diagnostics;
+using System.Text.Formatting;
+using System.Text.Utf8;
+
+namespace System.Text.Parsing
+{
+    public static partial class InvariantParser
+    {
+        /// <summary>
+        /// Parses an unsigned 64-bit integer from a location within a UTF-8 byte array buffer.
+        /// </summary>
+        /// <param name="utf8Text">The UTF-8 buffer, passed as a byte array.</param>
+        /// <param name="index">The index location of the value to be parsed within the buffer.</param>
+        /// <param name="value">The parsed value.</param>
+        /// <param name="bytesConsumed">The length (in bytes) of the unparsed value within the UTF-8 buffer.</param>
+        /// <returns>True if parsing is successful; false otherwise.</returns>
+		public static bool TryParse(byte[] utf8Text, int index, out long value, out int bytesConsumed)
+        {
+            Precondition.Require(utf8Text.Length > 0);
+
+            value = 0;
+            bytesConsumed = 0;
+            bool negative = false;
+
+            if (utf8Text[index] == '-')
+            {
+                negative = true;
+                index++;
+                bytesConsumed++;
+            }
+
+            for (int byteIndex = index; byteIndex < utf8Text.Length; byteIndex++) // loop through the byte array
+            {
+                if (utf8Text[byteIndex] < '0' || utf8Text[byteIndex] > '9') // if the next character is not a digit
```

Comment:
It might be better to read the next byte into a local, especially if we ever port this code to the out of band (i.e. corfxlab) Span<T>, which might not be as optimized as arrays are.